WordPress/gutenberg
# Editor: Introduce PluginPostExcerpt slot

## What?


This PR introduces the `<PluginPostExcerpt /> `slot component to extend the Post Excerpt panel.

 ```jsx
 import { __ } from '@wordpress/i18n';
 import { PluginPostExcerpt } from '@wordpress/edit-post';
 
const MyPluginPostExcerpt = () => (
  	<PluginPostExcerpt
 		className="my-plugin-post-excerpt"
 	>
 		{ __( 'Post excerpt custom content' ) }
 	</PluginPostExcerpt>
 );
 ```


## Why?


Because I'd like to be able to extend the excerpt panel. For instance, this Jetpack plugin PR enhances the panel with AI Assistance: https://github.com/Automattic/jetpack/pull/33529

## How?


By using the [Slot&Fill core API](https://developer.wordpress.org/block-editor/reference-guides/slotfills/#how-do-they-work) \m/

## Testing Instructions





* Go to the block editor in your dev-env 
* Open the console dev tool of your client
* Confirm the Excerpt panel works as usual, as expected.
* Create a custom component to be used to fill the excerpt slot, e.g.:

```js
function MyPluginPostExcerpt() {
	return React.createElement(
		wp.editPost.__experimentalPluginPostExcerpt,
		{
			className: 'my-plugin-post-excerpt',
		},
		'🎉 Post excerpt custom content 🎇'
	)
}
```
* Register the plugin, passing the component to be rendered

```js
wp.plugins.registerPlugin( 'my-custom-excerpt-content', {
    render: MyPluginPostExcerpt,
} );
```
<img width="496" alt="image" src="https://github.com/WordPress/gutenberg/assets/77539/d3a52a28-9bab-4a3d-9391-fd43bd9a3dfe">

* Confirm the excerpt panel has been extended with the component defined above (MyPluginPostExcerpt):

<img width="347" alt="Screenshot 2023-10-10 at 14 37 05" src="https://github.com/WordPress/gutenberg/assets/77539/037f77b3-9b59-45ed-8aa9-cd58672f6099">

### Testing Instructions for Keyboard


## Screenshots or screencast
